components/encryption/Cargo.toml

```diff
@@ -6,9 +6,10 @@ publish = false
 
 [features]
 failpoints = ["fail/failpoints"]
+openssl-vendored = ["openssl/vendored"]
 # openssl/vendored is necssary in order to conditionally building SM4 encryption
 # support, as SM4 is disabled on various openssl distributions, such as Rocky Linux 9.
-sm4 = ["openssl/vendored"]
+sm4 = ["openssl-vendored"]
```

Comment:
TiFlash should explicitly enables the "sm4" feature, otherwise upgrade may fail if user enables encryptio-at-rest sm4 before.

Reply:
I will also make sm4 a explicit feature